src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -1623,19 +1623,15 @@ protected boolean isUnitPlacementRestrictions() {
   protected boolean isPlayerAllowedToPlacementAnyTerritoryOwnedLand(final PlayerID player) {
     if (isPlaceInAnyTerritory()) {
       final RulesAttachment ra = (RulesAttachment) player.getAttachment(Constants.RULES_ATTACHMENT_NAME);
-      if (ra != null && ra.getPlacementAnyTerritory()) {
-        return true;
-      }
+      return ra != null && ra.getPlacementAnyTerritory();
```

Comment:
Not sure if this works, but perhaps this statement can be combined with the Boolean expression in the wrapping if clause if the rules attachment doesn't depend on the result of this expression.